File: gwwz/src/com/control/admin/ManageCustomerInfo.java
package com.control.admin;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import com.dao.CustomerInfoDAO;
import com.util.PublicToolCheckParam;
import com.util.PublicToolShowCustomerInfoList;

public class ManageCustomerInfo extends HttpServlet {

	/**
	 * 
	 */
	private static final long serialVersionUID = 3099608914702213565L;
	static CustomerInfoDAO cdao = new CustomerInfoDAO();

	//冻结
	public  static void coolCustomerInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
			String customerId = request.getParameter("customerId");
			String isCooled = request.getParameter("isCooled");
			if(PublicToolCheckParam.checkNullAndEmpty(customerId) && PublicToolCheckParam.checkNullAndEmpty(isCooled)){
					cdao.coolCustomer(Integer.parseInt(customerId),Integer.parseInt(isCooled));
					String isSearching = request.getParameter("isSearching");
					String currentPage = request.getParameter("currentPage");
					if(PublicToolCheckParam.checkNullAndEmpty(isSearching)){
						String key = request.getParameter("key");
						PublicToolShowCustomerInfoList.searchCustomerInfo(request, response, currentPage, key);
						request.setAttribute("key", key);
						request.setAttribute("isSearching", "yes");
					}
					else{
						PublicToolShowCustomerInfoList.showCustomerInfoList(request, response, currentPage);
					}
			}
			request.getRequestDispatcher("../admin/manageCustomerInfo.jsp").forward(request, response);
			
	}

	public static void coolMoreCustomerInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
			String currentPage = request.getParameter("currentPage");
			String[] ckb = request.getParameterValues("ckb");
			if(ckb != null){
				for(int i=0;i<ckb.length;i++){
	//				cdao.coolCustomer(Integer.parseInt(ckb[i]));
				}
			}
			PublicToolShowCustomerInfoList.showCustomerInfoList(request, response, currentPage);
			request.getRequestDispatcher("../admin/manageCustomerInfo.jsp").forward(request, response);
		
	}

	public static void searchCustomerInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
			String currentPage = request.getParameter("currentPage");
			String key = request.getParameter("key");
			PublicToolShowCustomerInfoList.searchCustomerInfo(request, response, currentPage, key);
			request.setAttribute("isSearching", "yes");
			request.setAttribute("key", key);
	
				CustomerInfoDAO  cdao = new CustomerInfoDAO();
		
			int dongjie = cdao.getsdongjie(key);
			int zhengchang = cdao.getszhengchang(key);
			request.setAttribute("dongjie", dongjie);
			request.setAttribute("zhengchang", zhengchang);

			request.getRequestDispatcher("../admin/manageCustomerInfo.jsp").forward(request, response);
		
	}

	public static void showManageCustomerInfoPage(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
		
		
			CustomerInfoDAO  cdao = new CustomerInfoDAO();
		
			int dongjie = cdao.getdongjie();
			int zhengchang = cdao.getzhengchang();
			request.setAttribute("dongjie", dongjie);
			request.setAttribute("zhengchang", zhengchang);

			String currentPage = request.getParameter("currentPage");
			PublicToolShowCustomerInfoList.showCustomerInfoList(request, response, currentPage);
			request.getRequestDispatcher("../admin/manageCustomerInfo.jsp").forward(request, response);
	
	}


/**
 * Constructor of the object.
 */
public ManageCustomerInfo() {
	super();
}


/**
 * Destruction of the servlet. <br>
 */
public void destroy() {
	super.destroy(); // Just puts "destroy" string in log
	// Put your code here
}

/**
 * The doGet method of the servlet. <br>
 *
 * This method is called when a form has its tag value method equals to get.
 * 
 * @param request the request send by the client to the server
 * @param response the response send by the server to the client
 * @throws ServletException if an error occurred
 * @throws IOException if an error occurred
 */
public void doGet(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {

	doPost(request, response);
}

/**
 * The doPost method of the servlet. <br>
 *
 * This method is called when a form has its tag value method equals to post.
 * 
 * @param request the request send by the client to the server
 * @param response the response send by the server to the client
 * @throws ServletException if an error occurred
 * @throws IOException if an error occurred
 */
public void doPost(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {


String method = request.getParameter("method");
if(PublicToolCheckParam.checkNullAndEmpty(method)){
	if(method.equals("showManageCustomerInfoPage")){
		showManageCustomerInfoPage(request, response);
	}
	else if(method.equals("cool")){
		coolCustomerInfo(request, response);
	}
	else if(method.equals("coolMore")){
		coolMoreCustomerInfo(request, response);
	}
	else if(method.equals("search")){
		searchCustomerInfo(request, response);
	}
}
}

	/**
	 * Initialization of the servlet. <br>
	 *
	 * @throws ServletException if an error occurs
	 */
	public void init() throws ServletException {
		// Put your code here
	}

}

File: gwwz/WebRoot/js/admin/manageCustomerInfo.js
function coolMoreCustomerInfo(){
	var ckb = document.getElementsByName("ckb");
	var flag = 0;
	var i;
	if(ckb.length == 0){
		alert("当前无任何客户信息!");
		return false;
	}
	else if(ckb.length > 0){
		for(i=0;i<ckb.length;i++){
			if(ckb[i].checked == true){
				flag = 1;
				break;
			}
		}
	}
	if(flag == 1){
		if(confirm("您确定要冻结当前所选客户?")){
			return true;
		}
		return false;
		
	}
	else{
		alert("请选择您要冻结的客户!");
		return false;
	}
}


function selectAll(){
	var ckb = document.getElementsByName("ckb");
	if(ckb.length > 0){
		if(document.getElementById("checkAll").checked == true){
			for(var i=0;i<ckb.length;i++){
				ckb[i].checked = true;
			}
		}
		else if(document.getElementById("checkAll").checked == false){
			for(var i=0;i<ckb.length;i++){
				ckb[i].checked = false;
			}
		}
	}
}



function searchCustomerInfo(){
	var key = document.getElementById("key").value;
	if(key == ""){
		alert("请输入查询关键词!");
		return false;
	}
	else{
		window.location.href = "../servlet/ManageCustomerInfo?method=search&key="+key+"";
		return true;
	}
}

function coolCustomerInfoOfSearching(customerId,isCooled,key){
	if(isCooled == "0"){
		if(confirm("您确定要冻结当前所选客户?")){
			window.location.href = "../servlet/ManageCustomerInfo?method=cool&customerId="+customerId+"&isCooled="+isCooled+"&isSearching=yes&key="+key+"";
		}
	}
	else{
		if(confirm("您确定要激活当前所选客户?")){
			window.location.href = "../servlet/ManageCustomerInfo?method=cool&customerId="+customerId+"&isCooled="+isCooled+"&isSearching=yes&key="+key+"";
		}
	}
}

function coolCustomerInfo(customerId,isCooled){
	if(isCooled == "0"){
		if(confirm("您确定要冻结当前所选客户?")){
			window.location.href = "../servlet/ManageCustomerInfo?method=cool&customerId="+customerId+"&isCooled="+isCooled+"";
		}
	}
	else{
		if(confirm("您确定要激活当前所选客户?")){
			window.location.href = "../servlet/ManageCustomerInfo?method=cool&customerId="+customerId+"&isCooled="+isCooled+"";
		}
	}
}




function back(){

	window.location.href = "../servlet/ManageCustomerInfo?method=showManageCustomerInfoPage";
	}
	
	